### ISSUE TITLE
Workflow for adding relations with the ribbon UI

### ISSUE THREAD

**Issue by quincylvania** [openstreetmap/iD#6551]
We could use the ribbon UI as an easier way to add new relations. For instance, adding a new bus route could work like this:

1. Search "bus route" with the Add Feature tool
2. Select the Bus Route preset
3. Enter a new mode where clicking on features adds them to the relation
    - See also #6465
4. Click all desired members.
    - We could limit which features are eligible, e.g. only allow roads and stops
5. Click Finish once done

Related: #4915

**Comment by BjornRasmussen** [comment]
This seems a lot like #1575

**Comment by DujaOSM** [comment]
Curious: has the work on this commenced?

To me it seems simpler to build the functionality around the current multi-selection editing, similar to tag editing in v3. For selected objects, present a list of existing relations on them and offer the user to add them to a new relation, an existing relation or remove from a relation. Having a special mode like you proposed above is generally a pain to develop and foolproof, and I don't think it's superior usability-wise.

**Comment by SilentSpike** [comment]
@DujaOSM I think the OP sounds more intuitive for the new workflow (user selects a feature rather than a geometry to add). However, your idea could certainly also be implemented as a means of adding to existing relations.

**Comment by DujaOSM** [comment]
[Modes are Evil](https://en.wikipedia.org/wiki/Mode_(user_interface)#Mode_errors). 

I would rather have relations more "equal in standing" to points and ways, i.e. treat them as independent objects within the UI rather than strictly attached to map objects. Along the following lines:

- Add the fourth geometry, "Relation", to the toolbar, after "Point", "Way" and "Area". When selected, the sidebar appears and the user must select the relation type, select an appropriate preset, and any additional tags. 
- For the time being, the relation has no members, and an error/warning/prompt appears to that effect. (Empty relations should not be allowed to be committed, IMO.)
- All newly created relations are permanently cached during the session and thus available for expansion (that avoids the issue when you scroll the view out of the target relation's current scope, so you cannot add a remote member to an existing relation)
- To add a new member to a relation, select one (as it is now) or more (as I propose) objects from the map and add them to the relation selected in the drop-down, pretty much the same as now. (No clumsy "New relation..." item, which breaks the workflow).

**Comment by DujaOSM** [comment]
Hey @quincylvania, when did you sneak this in? 2.19.5 I assume? I don't see it documented in release notes:
![MultiRelations](https://user-images.githubusercontent.com/48584318/103776817-4b7bf800-5030-11eb-90dc-08ccc669142c.png)
This resolves the last, and the most important, item from my list above. Kudos!

**Comment by quincylvania** [comment]
> Hey @quincylvania, when did you sneak this in? 2.19.5 I assume? I don't see it documented in release notes

@DujaOSM Yes, this is #7753! It was released in [2.19.0](https://github.com/openstreetmap/iD/blob/develop/CHANGELOG.md#2190) under Usability & Accessibility but I didn't make a big deal about it since it's sort of advanced and there was enough else to discuss.

**Comment by Mashin6** [comment]
I also think that relations should be a fourth button on the top ribbon. There is no reason not to, since they are standard OSM objects. How I imagine it could work:

1. Press 4 or click relation button.
2. iD goes from drawing mode to selection-only mode where user can select multiple objects.
3. In the sidebar will already be suggestions what type of relation they want to create.
4. Select e.g. multipolygon and iD automatically determines what is outer and inner parts, warns about bad members (in this case unclosed ways, points, touching outer rings.. etc.)
5. Sidebar shows only relevant fields e.g. hiking route will have colour and osmc:symbol
6. User can still keep selecting and unselecting objects and iD adds them or removes them
8. User confirms changes

**Comment by k-yle** [comment]
I'm going to close this issue, since it's an issue in the v3 prototype version, which is (sadly) abandoned and [no longer deployed](https://github.com/openstreetmap/iD/commit/36f4eeb7e24ba3ffd16f0d48780f3c56d39f2bd1).

---

lets continue the discussion in the more recent issues #9359 and #4915
